src/components/ModelingMachineProvider.tsx

```diff
@@ -170,10 +163,23 @@ export const ModelingMachineProvider = ({
           })()
         },
         'Set mouse state': assign({
-          mouseState: (_, event) => event.data,
-          segmentHoverMap: ({ mouseState, segmentHoverMap }, event) => {
-            if (event.data.type === 'isHovering') {
-              const parent = getParentGroup(event.data.on, [
+          mouseState: ({ event }) =>
+            (
+              event as Extract<
+                ModelingMachineEvent,
+                { type: 'Set mouse state' }
+              >
+            ).data,
+          segmentHoverMap: ({
+            context: { mouseState, segmentHoverMap },
+            event,
+          }) => {
+            const _event = event as Extract<
+              ModelingMachineEvent,
+              { type: 'Set mouse state' }
+            >
```

Comment:
I wish there were a way for these events to have the correct type without `as`. We were hoping to forbid `as` using a lint in the future.

Reply:
We could narrow the event object by guarding against `event.type`.

Reply:
Pushed it.